sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,366 @@
+"""
+This module implements STEREO Client.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import re
+
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+
+__all__ = ['SECCHIClient', 'PLASTICClient', 'IMPACTClient', 'SWAVESClient']
+
+class SECCHIClient(GenericClient):
+    """
+    Returns a list of URLS to SECCHI files corresponding to value of input timerange.
+    URL source: `http://stereo-ssc.nascom.nasa.gov/data/beacon/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument = 'secchi'
+
+    Source: Two arguments, case-sensitive - 'STEREO_A' or 'STEREO_B'.
+
+    Detector: Can take any four arguments, case-insensitive - 'euvi', 'cor2', 'hi_1', 'hi_2'
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(a.Time('2007/3/2 17:00:00', '2007/3/3 17:15:00'), a.Instrument('secchi'),
+        a.Source('STEREO_A'), a.Detector('euvi'))
+    >>> print(results)
+        [<Table length=4>
+        Start Time           End Time      Source Instrument
+           str19               str19         str5     str6
+    ------------------- ------------------- ------ ----------
+    2007-03-02 00:00:00 2007-03-03 00:00:00  ahead     secchi
+    2007-03-03 00:00:00 2007-03-04 00:00:00  ahead     secchi
+    2007-03-04 00:00:00 2007-03-05 00:00:00  ahead     secchi
+    2007-03-05 00:00:00 2007-03-06 00:00:00  ahead     secchi]
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        if not timerange:
+            return []
+        regex = re.compile('[^a-zA-Z0-9]')
+        source = kwargs.get('source')
+        table_ = {'STEREO_A':'ahead', 'STEREO_B':'behind'}
+        detector = regex.sub('', kwargs.get('detector')).lower() #default to euvi ?
+        table = {'euvi':'euvi', 'hi1':'hi_1', 'hi2':'hi_2', 'cor2':'cor2'}
+        #table here just simplifies life.
+        #Would have to include 2 or 3 if statements to handle hi1, hi2 and the other two.
+        #Instead I just use a dictionary.
+        prefix = 'http://stereo-ssc.nascom.nasa.gov/data/beacon/{source}/secchi/img/{det}/%Y%m%d/%Y%m%d_%H%M%S_'
```

Comment:
This points to beacon data, very low quality but close-to-real time data.  Is this is what was intended for this client?